README.md

```diff
@@ -56,186 +50,318 @@ npm install @dojo/has
+	protected render() {
+		const { onItemClick, items } = this.properties;
 
-w('my-widget', properties);
-w('my-widget', properties, children);
+		return v('ul', { classes: { list: true } }, items.map(({ id, content, highlighted }) => {
+			return w(ListItem, { key:id, content, highlighted, onItemClick });
+		});
+	}
+}
 ```
 
-The example above that uses a string for the `widgetConstructor `, is taking advantage of the [widget registry](#widget-registry) functionality.
-The widget registry allows for the lazy loading of widgets.
+Additionally the `ListItem` is now reusable in other areas of our application(s).
 
-### tsx
+### Mixins
 
-In additional to the programatic functions `v` and `w`, widget-core optionally supports the use of the `jsx` syntax known as [`tsx`](https://www.typescriptlang.org/docs/handbook/jsx.html) in TypeScript.
+Dojo 2 makes use of mixins to decorate additional functionality and properties to existing widgets. Mixins provide a mechanism that allows loosely coupled design and composition of behaviors into existing widgets without having to change the base widget.
 
-To start to use `jsx` in your project the widgets need to be named with a `.tsx` extension and some configuration is required in the project's `tsconfig.json`:
+TypeScript supports mixins using a constructor type of `new (...args: any[]) => any;` that enables a class to be passed as a function argument and extended to add new functionality.
 
-Add the configuration options for `jsx`:
+Example mixin that adds method `setState` and `readonly` `state` property:
 
-```
-"jsx": "react",
-"jsxFactory": "tsx",
-```
+```ts
+// interface for the extended API
+interface StateMixin {
+	readonly state: Readonly<any>;
+	setState(state: any): void;
+}
 
-Include `.tsx` files in the project:
+// function that accepts a class that extends WidgetBase and returns the extended class with the StateMixin
+// behavior
+function StateMixin<T extends new(...args: any[]) => WidgetBase>(Base: T): T & (new(...args: any[]) => StateMixin) {
+	return class extends Base {
+		private _state: any;
 
-```
- "include": [
- 	"./src/**/*.ts",
- 	"./src/**/*.tsx"
- ]
+		public setState(state: any): void {
+			// shallow copy of the state
+			this._state = { ...this._state, ...state };
+			// invalidate the widget
+			this.invalidate();
+		}
+
+		public get state(): any {
+			return this._state;
+		}
+	};
+}
 ```
 
-Once the project is configured, `tsx` can be used in a widget's `render` function simply by importing the `tsx` function as `import { tsx } from '@dojo/widget-core/tsx';`
+Examples of Dojo 2 mixins can be seen with `ThemeableMixin` and `I18nMixin` that are described in [Classes & theming](#classes--theming) and [Internationalization](#internationalization) sections.
 
-```tsx
-class MyWidgetWithTsx extends WidgetBase<MyProperties> {
-	protected render(): DNode {
-		const { clear, properties: { completed, count, activeCount, activeFilter } } = this;
+### Classes & Theming
```

Comment:
Styling and Theming (or preface with CSS Classes, to avoid confusion with ES Classes?)